packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -125,31 +209,31 @@ class _AnimatedChildSwitcherState extends State<AnimatedChildSwitcher> with Tick
     super.dispose();
   }
 
+  bool get hasNewChild => widget.child != null;
+  bool get hasOldChild => _currentChild?.widget != null;
```

Comment:
looks like _currentChild can never be false, since we set it in initState, so you can remove the `?`

Reply:
unless you change the semantics as suggested above

Reply:
in which case you could just make this _currentChild != null

then you could use it directly in addEntry

Reply:
Yeah, I updated it to the new semantics.